stl/inc/numeric

```diff
@@ -612,7 +612,6 @@ _NODISCARD constexpr common_type_t<_Mt, _Nt> gcd(const _Mt _Mx, const _Nt _Nx) n
         const auto _Mx_trailing_zeroes = static_cast<unsigned long>(_Countr_zero_impl(_Mx_magnitude));
         const auto _Common_factors_of_2 =
             (_STD min)(_Mx_trailing_zeroes, static_cast<unsigned long>(_Countr_zero_impl(_Nx_magnitude)));
-        _Nx_magnitude >>= _Common_factors_of_2;

```

Comment:
I don't see why this is redundant; three lines down, we shift `_Nx_magnitude` down by `countr_zero(_Nx_magnitude)`, so by left line 619,

- `_Nx_magnitude = old _Nx_magnitude >> (_Common_factors_of_2 + countr_zero(_Nx_magnitude >> _Common_factors_of_2))`....

oh, I see it now. Okay, now that we don't do this shift, we can cache `_Countr_zero_impl(_Nx_magnitude)` and use it twice; that would result in even better performance (in debug at least), and we wouldn't have to keep around `_Common_factors_of_2`; we can just calculate it in right line 626.

Reply:
Note that reusing `_Countr_zero_impl(_Nx_magnitude)` for the first iteration still results in a redundant `_Countr_zero_impl` for the last iteration versus a redundant `_Countr_zero_impl` for the first iteration.

Reply:
```c++
const auto _Mx_trailing_zeroes = static_cast<unsigned long>(_Countr_zero_impl(_Mx_magnitude));
auto _Nx_trailing_zeroes       = static_cast<unsigned long>(_Countr_zero_impl(_Nx_magnitude));
const auto _Common_factors_of_2 = (_STD min)(_Mx_trailing_zeroes, _Nx_trailing_zeroes);
_Mx_magnitude >>= _Mx_trailing_zeroes;
do {
    _Nx_magnitude >>= _Nx_trailing_zeroes;
    if (_Mx_magnitude > _Nx_magnitude) {
        _Common_unsigned _Temp = _Mx_magnitude;
        _Mx_magnitude          = _Nx_magnitude;
        _Nx_magnitude          = _Temp;
    }

    _Nx_magnitude -= _Mx_magnitude;
    _Nx_trailing_zeroes = static_cast<unsigned long>(_Countr_zero_impl(_Nx_magnitude)); 
} while (_Nx_magnitude != 0U);
```

Reply:
I don't understand why... oh, never mind, I understand now. No changes requested; thanks!